homeassistant/components/samsungtv/bridge.py

```diff
@@ -0,0 +1,263 @@
+"""samsungctl and samsungtvws bridge classes."""
+from abc import ABC, abstractmethod
+
+from samsungctl import Remote
+from samsungctl.exceptions import AccessDenied, ConnectionClosed, UnhandledResponse
+from samsungtvws import SamsungTVWS
+from samsungtvws.exceptions import ConnectionFailure
+from websocket import WebSocketException
+
+from homeassistant.const import (
+    CONF_HOST,
+    CONF_ID,
+    CONF_METHOD,
+    CONF_NAME,
+    CONF_PORT,
+    CONF_TIMEOUT,
+    CONF_TOKEN,
+)
+
+from .const import (
+    CONF_DESCRIPTION,
+    LOGGER,
+    METHOD_LEGACY,
+    RESULT_AUTH_MISSING,
+    RESULT_NOT_SUCCESSFUL,
+    RESULT_NOT_SUPPORTED,
+    RESULT_SUCCESS,
+    VALUE_CONF_ID,
+    VALUE_CONF_NAME,
+)
+
+
+class SamsungTVBridge(ABC):
+    """The Base Bridge abstract class."""
+
+    @staticmethod
+    def get_bridge(method, host, port=None, token=None):
+        """Get Bridge instance."""
+        if method == METHOD_LEGACY:
+            return SamsungTVLegacyBridge(method, host, port)
```

Comment:
Shouldn't we set port to the legacy port? It can only be one thing, right?

Reply:
In a previous review @escoand said that samsungctl doesn't need the port parameter for legacy devices. That's why I removed it.

Reply:
Ok, yeah it will default to 55000 in the library. It's a but awkward that its still needed as an item in the config, but that's not our fault.

https://github.com/Ape/samsungctl/blob/81cf14aa138c32ff10aaa424864d6fc92b987322/samsungctl/remote_legacy.py#L9-L14